src/Telegram.Bot/Requests/SetWebhookRequest.cs

```diff
@@ -0,0 +1,115 @@
+namespace Telegram.Bot.Requests
+{
+    /// <summary>
+    /// Specify a url and receive incoming updates via an outgoing webhook
+    /// </summary>
+    public class SetWebhookRequest : RequestBase<bool>
+    {
+        private IEnumerable<UpdateType> _allowedUpdateTypes;
+
+        /// <summary>
+        /// HTTPS url to send updates to. Use an empty string to remove webhook integration.
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string Url { get; set; }
+
+        /// <summary>
+        /// Public key certificate so that the root certificate in use can be checked
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public Stream Certificate { get; set; }
+
+        /// <summary>
+        /// Maximum allowed number of simultaneous HTTPS connections to the webhook for update delivery
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public int MaxConnections { get; set; }
+
+        /// <summary>
+        /// List the types of updates you want your bot to receive
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public IEnumerable<UpdateType> AllowedUpdates
+        {
+            get => _allowedUpdateTypes;
+            set
+            {
+                if (value != null && value.All(u => u != UpdateType.All))
+                {
+                    _allowedUpdateTypes = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new request with uri
+        /// </summary>
+        /// <param name="url">HTTPS url to send updates to</param>
+        public SetWebhookRequest(string url)
+            : base("setWebhook")
+        {
+            Url = url;
+        }
+
+        /// <summary>
+        /// Initializes a new request with uri and certificate
+        /// </summary>
+        /// <param name="url">HTTPS url to send updates to</param>
+        /// <param name="certificate">Public key certificate so that the root certificate in use can be checked</param>
+        public SetWebhookRequest(string url, Stream certificate)
+            : base("setWebhook")
+        {
+            Url = url;
+            Certificate = certificate;
+        }
+
+        /// <summary>
+        /// Initializes a new request
+        /// </summary>
+        public SetWebhookRequest()
+            : base("setWebhook")
+        { }
+
+        /// <summary>
+        /// Generate content of HTTP message
+        /// </summary>
+        /// <param name="serializerSettings">JSON serialization setting</param>
+        /// <returns>Content of HTTP request</returns>
+        public override HttpContent ToHttpContent(JsonSerializerSettings serializerSettings)
+        {
+            var multipartContent = new MultipartFormDataContent(Guid.NewGuid().ToString() + DateTime.UtcNow.Ticks)
```

Comment:
I'm not really sure whether generating a new GUID every single time for mutlipart requests is a good idea. I started doing it before but we might want to reassess this approach.

Reply:
We can use predefined string to be used as boundary in every request with multipart/form-data.

Reply:
having a constant value is a good idea but what unique string could we use that isn't in another part of the payload?

Reply:
I guess any random string would be sufficient. We can generate GUID somewhere inside internal class in a static constructor so it will be unique on every application start. How high is the chance of a collision with something inside of the payload during single app lifetime cycle?

Reply:
Though constant value even during single app cycle could lead to greater chance of a collision. I think generating the boundary on every request is not that bad. What do you think is more performant: using Guid or generating random string using preallocated StringBuilder that is cleared after every ToString call?

Something like:
```
builder.Clear();
var rnd = new Random();
...
builder.Append((char)(rnd.Next(65, 90)))
return builder.ToString();
```